ldms/src/sampler/file_exporter/ldms-sampler_file_exporter.rst

```diff
@@ -0,0 +1,239 @@
+.. _file_exporter
+
+===============
+ file_exporter
+===============
+
+-------------------------------------------
+Man page for the LDMS file_exporter plugin
+-------------------------------------------
+
+:Date: 25 Aug 2025
+:Manual section: 7
+:Manual group: LDMS Sampler
+
+SYNOPSIS
+========
+
+| Within ldmsd_controller or a configuration file:
+.. code-block:: text
+
+   config name=CFG_NAME name=file_exporter [ <attr>=<valiue> ]
+
+DESCRIPTION
+===========
+
+With LDMS (Lightweight Distributed Metric Service), plugins for the
+**ldmsd** (LDMS Daemon) are configured via **ldmsd_controller** or a
+configuration file. The *file_exporter* plugin searches a directory
+and imports all files in the directory into metric sets. Each
+file found results in its own schema and metric set instance.
+
+The rules governing how the contents of a file are imported into a
+metric set are specified in the imported file itself.
+
+This plugin is multi-instance capable.
+
+CONFIGURATION ATTRIBUTE SYNTAX
+==============================
+
+The file_exporter plugin uses the *sampler_base* base class. This man
+page covers only the configuration attributes specific to this plugin,
+or those for which the behavior is different from that of
+:ref:`ldms_sampler_base(7) <ldms_sampler_base>`.
+
+**config**\  **name**\ =<cfg_name> **plugin**\ =file_exporter **dir_path**\ =<PATH>
+**schema**\ =<schema-format> **instance**\ =<instance-format> **uid**\ =<user-id> **gid**\ =<group-id>
+*<SAMPLER_BASE_OPTIONS>*
+
+
+Description
+-----------
+
+   **name**\ *=<config_name>*
+      The name of the configuration. If this is specified without **plugin** it assumed that **plugin** == **name**
+
+   **plugin**\ *=file_exporter*
+      This must be *file_exporter*
+
+   **dir_path**\ *=<PATH>*
+      The path to the directory from which all files will be imported.
+
+   **uid**\ *=<user-id>*
+      The user-id that will be placed in the meta-data of the metric set. This is only
+      honored if the **ldmsd** is owned by root, otherwise it is the value returned
```

Comment:
Why? geteuid() can return anything a user wants it to return. We can't trust it, so why bother restricting this field? It can't be trusted if the daemon isn't trusted.